tensorflow/python/eager/function.py

```diff
@@ -2902,6 +2902,8 @@ def __init__(self,
     self._function_attributes = attributes or {}
     self._capture_by_value = capture_by_value
     self.tracing_count = 0
+    self._hashable_input_signature = _make_input_signature_hashable(
+        self.flat_input_signature)
```

Comment:
we should only do this when the self.input_signature is not None

Reply:
Done